From 1e42191d69ce19dfe37ec3fd9d56e335a6eb1f46 Mon Sep 17 00:00:00 2001
From: Sebastian Panceac <sebastian@balena.io>
Date: Mon, 17 Dec 2018 16:58:41 +0100
Subject: [PATCH] Load splash screen from emmc when booting from NAND

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Sebastian Panceac <sebastian@balena.io>
---
diff --git a/board/variscite/mx6var_som/mx6var_som.c b/board/variscite/mx6var_som/mx6var_som.c
index a892c94c518..157e7ac3e8c 100644
--- a/board/variscite/mx6var_som/mx6var_som.c
+++ b/board/variscite/mx6var_som/mx6var_som.c
@@ -511,7 +511,7 @@ static void set_splashsource_to_boot_rootfs(void)
 		return;
 
 #ifdef CONFIG_NAND_BOOT
-	env_set("splashsource", env_get("rootfs_device"));
+	env_set("splashsource", "emmc");
 #else
 	if (mmc_get_env_dev() == 2)
 		env_set("splashsource", "emmc");
