From 57b667e9ab792d37f319b4aace9315ab950e9f6b Mon Sep 17 00:00:00 2001
From: Sebastian Panceac <sebastian@balena.io>
Date: Mon, 17 Dec 2018 15:23:06 +0100
Subject: [PATCH] Load kernel and rootfs from MMC when booting from NAND

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Sebastian Panceac <sebastian@balena.io>
---
 include/configs/mx6var_som.h | 37 +++++++-----------------------------
 1 file changed, 7 insertions(+), 30 deletions(-)

diff --git a/include/configs/mx6var_som.h b/include/configs/mx6var_som.h
index 573b94e0dd..13f7f6bcd6 100644
--- a/include/configs/mx6var_som.h
+++ b/include/configs/mx6var_som.h
@@ -57,7 +57,11 @@
 /* MMC Configs */
 #define CONFIG_SYS_FSL_ESDHC_ADDR	0
 #define CONFIG_SYS_FSL_USDHC_NUM	2
+#ifndef CONFIG_NAND_BOOT
 #define CONFIG_SYS_MMC_ENV_DEV		0
+#else
+#define CONFIG_SYS_MMC_ENV_DEV		1
+#endif
 #ifndef CONFIG_SYS_MMC_ENV_PART
 #define CONFIG_SYS_MMC_ENV_PART		0
 #endif
@@ -152,35 +156,8 @@
 	"else run netboot; fi;"
 
 #ifdef CONFIG_NAND_BOOT
-#define NAND_BOOT_ENV_SETTINGS \
-	"nandargs=setenv bootargs console=${console},${baudrate} ubi.mtd=3 " \
-		"root=ubi0:rootfs rootfstype=ubifs\0" \
-	"rootfs_device=nand\0" \
-	"boot_device=nand\0" \
-	"nandboot=nand read ${loadaddr} 0x400000 0x800000; " \
-		"nand read ${fdt_addr} 0x3e0000 0x20000; " \
-		"bootm ${loadaddr} - ${fdt_addr};\0" \
-	"bootcmd=" \
-		"if test ${rootfs_device} != emmc; then " \
-			"run nandargs; " \
-			"run videoargs; " \
-			"run optargs; " \
-			"echo booting from nand ...; " \
-			"run nandboot; " \
-		"else " \
-			"if test ${boot_device} != emmc; then " \
-				"run mmcargs; " \
-				"run videoargs; " \
-				"run optargs; " \
-				"echo booting from nand (rootfs on emmc)...; " \
-				"run nandboot; " \
-			"else " \
-				"setenv mmcdev 1; " \
-				MMC_BOOTCMD \
-			"fi; " \
-		"fi;\0" \
-	"mtdids=" MTDIDS_DEFAULT "\0" \
-	"mtdparts=" MTDPARTS_DEFAULT "\0"
+#define NAND_BOOT_ENV_SETTINGS ""
+#define CONFIG_BOOTCOMMAND MMC_BOOTCMD
 #else
 #define NAND_BOOT_ENV_SETTINGS ""
 #define CONFIG_BOOTCOMMAND	MMC_BOOTCMD
@@ -323,7 +300,7 @@
 
 #ifdef CONFIG_NAND_BOOT
 #define CONFIG_CMD_NAND
-#define CONFIG_ENV_IS_IN_NAND
+#define CONFIG_ENV_IS_IN_MMC
 #define CONFIG_SYS_NAND_U_BOOT_OFFS	0x200000
 #else
 #define CONFIG_ENV_IS_IN_MMC
-- 
2.17.1

