From 83c3bc27c9afb281d7615f65c03f22acce54aee7 Mon Sep 17 00:00:00 2001
From: Sebastian Panceac <sebastian@resin.io>
Date: Wed, 10 Oct 2018 15:40:50 +0200
Subject: [PATCH] Integrate Resin u-boot environment

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Sebastian Panceac <sebastian@resin.io>
---
diff --git a/configs/mx6var_som_sd_defconfig b/configs/mx6var_som_sd_defconfig
index bbd995d9042..bdba90a 100644
--- a/configs/mx6var_som_sd_defconfig
+++ b/configs/mx6var_som_sd_defconfig
@@ -6,7 +6,7 @@ CONFIG_SPL_GPIO=y
 CONFIG_SPL_LIBCOMMON_SUPPORT=y
 CONFIG_SPL_LIBGENERIC_SUPPORT=y
 CONFIG_NR_DRAM_BANKS=1
-CONFIG_ENV_SIZE=0x2000
+CONFIG_ENV_SIZE=0x4000
 CONFIG_ENV_OFFSET=0xC0000
 CONFIG_MX6QDL=y
 CONFIG_TARGET_MX6VAR_SOM=y
diff --git a/include/configs/mx6var_som.h b/include/configs/mx6var_som.h
index 9c8204082ec..a294894 100644
--- a/include/configs/mx6var_som.h
+++ b/include/configs/mx6var_som.h
@@ -53,7 +53,7 @@
 	"mmcbootpart=1\0" \
 	"mmcrootpart=" __stringify(MMC_ROOT_PART) "\0" \
 	"mmcargs=setenv bootargs console=${console},${baudrate} " \
-		"root=/dev/mmcblk${mmcblk}p${mmcrootpart} rootwait rw\0" \
+		"${resin_kernel_root} rootwait rw ${os_cmdline} \0" \
 	"loadbootenv=" \
 		"load mmc ${mmcdev}:${mmcbootpart} ${loadaddr} ${bootdir}/${bootenv};\0" \
 	"importbootenv=echo Importing bootenv from mmc ...; " \
--
2.17.1
